### PR TITLE
Move data retrieving and deserialization to kernel.request listeners

### PR DESCRIPTION
**Pull request by dunglas**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | todo

### CODE REVIEW

**Comment by vincentchalamon** [comment]
Please rebase from master, then 👍

**Comment by Simperfit** [comment]
👍

File: src/Bridge/Symfony/Bundle/Resources/config/doctrine_orm.xml [hunk 1]
```diff
@@ -105,13 +105,13 @@
             <argument>%api_platform.collection.pagination.enabled_parameter_name%</argument>
             <argument>%api_platform.collection.pagination.items_per_page_parameter_name%</argument>
 
-            <tag name="api_platform.doctrine.orm.query_extension.collection" priority="-20" />
+            <tag name="api_platform.doctrine.orm.query_extension.collection" priority="16" />
```
Comment (teohhanhui) [inline src/Bridge/Symfony/Bundle/Resources/config/doctrine_orm.xml]: This is a regression from #524. The OrderExtension must be run before the PaginationExtension.

Comment (teohhanhui) [inline src/Bridge/Symfony/Bundle/Resources/config/doctrine_orm.xml]: Or rather, `PaginationExtension` should be split into two. And there should be multiple passes / stages for the extensions.

Comment (soyuka) [inline src/Bridge/Symfony/Bundle/Resources/config/doctrine_orm.xml]: +1

**Pull request merged**